src/typing_extensions.py

```diff
@@ -3572,6 +3572,34 @@ def _raise_attribute_error(self, name: str) -> Never:
         def __repr__(self) -> str:
             return self.__name__
 
+        if sys.version_info < (3, 11):
+            def _check_single_param(self, param, recursion=0):
```

Comment:
Doesn't seem like this needs to be a generator, it always yields a single value. Let's make it return that value instead and simplify `_check_parameters`.

Reply:
Looking at it again. As the current cpython implementation does not not seems to use _type_convert or any restrictions, should we just drop all checks here?

Reply:
Changes in #489 are necessary to remove `_check_parameters` / `_type_convert`. More specifically, `_ConcatenateGenericAlias` needs `_copy_with` and `__getitem__` if it is not converted to a list via the above functions to be handled correctly in 3.8 & 3.9.